### PR TITLE
Update LEGA to v0.2.0

### LINKED ISSUE #9: Roll deployment on config update
### Description

We should roll the depoyment on configmap chages
### Proposed solution

Keep ttrack of the checksum for the CM i question and recreate the pod if the checksum changes.
### Definition of Done

Changes to a configmap will recreate all related pods.

### LINKED ISSUE #32: Add TLS option to the chart
### Definition of Done

TLS certificates are deployed an used by the containers.

### LINKED ISSUE #36: New profile for DataEdge
### Description
After the https://github.com/EGA-archive/ega-data-api/pull/135 is merged we need to add `LocalEGA` profile to DataEdge, and this issue is so that we don't forget

### Definition of Done
DataEdge has the new profile

### How to test

e2e test still pass.

### PR DESCRIPTION
**Pull request by jbygdell**
### Describe the pull request:

- [ ] Bug fix
- [X] Functional change
- [X] New feature
- [X] Code cleanup
- [ ] Build system change
- [X] Documentation change
- [ ] Language translation

### Pull request long description:

Update deployment to match v0.2.0 images, all connections are TLS terminated both external and internal.
For dataAPI an init container is used to convert the standard certificates to a java keystore.
Add ingress option to data out services.
Python services now uses an config file instead of ENVs
### Changes made:

1) Update the container images to v0.2.0
2) Add TLS to all services
3) Add Ingress for data out
4) Removed trailing whitespace at the end of some lines.
5) Change config s from ENV to ini file.

### Related issues:

Fixes #32 Fixes #36 Fixes #9 
### Documentation change:

The new config variables has been added to the read me file.

### CODE REVIEW

**Comment by jbygdell** [comment]
Tests will fail since tester is not updated to use https.

File: README.md [hunk 1]
```diff
@@ -117,6 +117,10 @@ Parameter | Description | Default
 `config.data_storage_s3_chunk_size` | S3 chunk size in MB | `16`
 `config.data_storage_location` | Path to FileStorage volume | `/ega/data_archive`
 `config.data_storage_mode` | File mode in storage volume | `2750`
+`config.tls_certs` | boolean to create secrets tha hold the TLS certificates | `true`
```
Comment (blankdots) [inline README.md:120]:
```suggestion
`config.tls_certs` | boolean to create secrets that hold the TLS certificates | `true`
```

File: README.md [hunk 2]
```diff
@@ -131,6 +135,11 @@ Parameter | Description | Default
 `secrets.s3_access_key` | Access key to S3 storage | `""`
 `secrets.s3_secret_key` | Secret key to S3 storage  | `""`
 `secrets.token`| jwt token for testing file download via dataedge | ``
+`ingress.deploy` | Create ingress for data out components when set to true | `false`
+`ingress.hostname` | The hostname that will be used in the ingress path | `""`
+`ingress.tls` | Boolean that controles wheter TLS should be enalbed for the ingress | `true`
```
Comment (blankdots) [inline README.md:140]:
```suggestion
`ingress.tls` | Boolean that controls whether TLS should be enabled for the ingress | `true`
```

File: README.md [hunk 3]
```diff
@@ -131,6 +135,11 @@ Parameter | Description | Default
...
+`ingress.secretName` | Name of the secret that holds the TLS certifiactes, the secret must be manuallly created unless cert-manager is installed | `""`
```
Comment (blankdots) [inline README.md:141]:
```suggestion
`ingress.secretName` | Name of the secret that holds the TLS certificates, the secret must be manually created unless cert-manager is installed | `""`
```

File: ega-charts/localega/templates/dataedge-deploy.yaml [hunk 4]
```diff
@@ -23,13 +23,26 @@ spec:
         role: dataedge
         release: {{ .Release.Name }}
     spec:
+      initContainers:
+      - name: tls-init
```
Comment (blankdots) [inline ega-charts/localega/templates/dataedge-deploy.yaml:27]: should I generate these in `lega-deploy-init` ?

File: ega-charts/localega/templates/mq-certs.yaml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+{{ if .Values.config.tls_certs }}
+{{- $cafile := .Files.Glob (printf "config/certs/%s" .Values.config.tls_ca_root_file) }}
+{{- $mqcerts := .Files.Glob "config/certs/mq*" }}
```
Comment (blankdots) [inline ega-charts/localega/templates/mq-certs.yaml:3]: for this we generate in `deploy-init` under `mq-server*`, but this is ok as well.

**Review by blankdots — APPROVED** [review]
So far looks good, just a few typos that need fixing.

Changes approved with the note that it has not yet been deployed and tested.

File: ega-charts/localega/templates/mq-certs.yaml (hunk 5 above)
Comment (jbygdell) [reply ega-charts/localega/templates/mq-certs.yaml:3]: Mq can act as both client and server so globing `mq-*` makes it future proof if we are creating client certs in the future.

Comment (blankdots) [reply ega-charts/localega/templates/mq-certs.yaml:3]: ok, then I will add to generate some client certs in `deploy-init`.

**Comment by blankdots** [comment]
Closed in favour of #41

**Pull request closed without merging**